src/analysis/asymptotics/asymptotics.lean

```diff
@@ -49,12 +49,19 @@ open_locale topological_space big_operators classical filter nnreal
 namespace asymptotics
 
 variables {α : Type*} {β : Type*} {E : Type*} {F : Type*} {G : Type*}
-  {E' : Type*} {F' : Type*} {G' : Type*} {R : Type*} {R' : Type*} {𝕜 : Type*} {𝕜' : Type*}
-
-variables [has_norm E] [has_norm F] [has_norm G] [normed_group E'] [normed_group F']
-  [normed_group G'] [normed_ring R] [normed_ring R'] [normed_field 𝕜] [normed_field 𝕜']
-  {c c' : ℝ} {f : α → E} {g : α → F} {k : α → G} {f' : α → E'} {g' : α → F'} {k' : α → G'}
-  {l l' : filter α}
+  {E' : Type*} {F' : Type*} {G' : Type*}
+  {E'' : Type*} {F'' : Type*} {G'' : Type*}
+  {R : Type*} {R' : Type*} {𝕜 : Type*} {𝕜' : Type*}
+
+variables [has_norm E] [has_norm F] [has_norm G]
+variables [semi_normed_group E'] [semi_normed_group F'] [semi_normed_group G']
+variables [normed_group E''] [normed_group F''] [normed_group G'']
```

Comment:
Is it possible to accomplish the same with some sectioning? I would prefer a `section has_norm/semi_normed_group/normed_group` over adding all these `'`.

Reply:
I think that would result in an awful diff. Presumably the same decision was made we when adding the first set of `'`s